UI/Qt: Fix hover_label hiding URLs

Sidicer opened the pull request:
`m_hover_label` did not have checks if the mouse is in the same location which caused clickable URLs to be hidden.

![image](https://github.com/user-attachments/assets/83467fa6-bddb-49e8-b011-87a9005e75e4)

Set max length for the `hover_label` text to be `50%` of the window width and added a check for mouse position. If there's a chance they might overlap - `hover_label` is moved to the opposite side of the window.
`Ladybird/Qt/WebContentView.cpp` fix provided by @tcl3 

https://github.com/user-attachments/assets/c0947ae8-93c2-4dc0-800a-2d23d498849a

_Moved the `hover_label` up by additional 4px if `find in page` is visible after recording the video_

ADKaster commented:
can you revert the changes to vcpkg.json? it's supposed to be a symlink

tcl3 commented on Ladybird/Qt/Tab.cpp:
```diff
@@ -893,15 +894,28 @@ void Tab::resizeEvent(QResizeEvent* event)
 
+void Tab::mouseMoveEvent(QMouseEvent* event)
+{
+    update_hover_label();
```
It feels a bit excessive to call this method on every mouse move event. I think we should be able to get away with calling this only when the mouse is over the label. To do this you could make `m_hover_label` a subclass of `QLabel` and override `enterEvent()` like this:

```c++
class HyperlinkLabel final : public QLabel {
    Q_OBJECT
public:
    explicit HyperlinkLabel(QWidget *parent = nullptr)
        : QLabel(parent)
    {
        setMouseTracking(true);
    }

    virtual void enterEvent(QEnterEvent* event) override
    {
        emit mouse_entered(event);
    }

signals:
    void mouse_entered(QEnterEvent*);
    
};
```

Then, in `Tab.cpp` you can call `update_label()` when the `mouse_entered` signal is fired like this:
```c++
    QObject::connect(m_hover_label, &HyperlinkLabel::mouse_entered, [this] {
        update_hover_label();
    });
```

Sidicer replied:
Yes, I also thought that it might even hinder the performance, but couldn't come up with a way. Thank you for the suggestion and indepth review

tcl3 commented on Ladybird/Qt/Tab.cpp:
```diff
@@ -893,15 +894,28 @@ void Tab::resizeEvent(QResizeEvent* event)
+    auto hover_label_height = height() - m_hover_label->height();
     if (m_find_in_page->isVisible())
-        hover_label_height -= m_find_in_page->height() - 4;
+        hover_label_height -= m_find_in_page->height() - 8;
```
This causes the label and the find in page widget to overlap on my machine:

![image](https://github.com/user-attachments/assets/56ca89f4-6307-4583-bc45-efccfd4fb5f0)

tcl3 commented on Ladybird/Qt/Tab.cpp:
```diff
@@ -893,15 +894,28 @@ void Tab::resizeEvent(QResizeEvent* event)
-        hover_label_height -= m_find_in_page->height() - 4;
+        hover_label_height -= m_find_in_page->height() - 8;
+
+    QPoint mousePosition = mapFromGlobal(QCursor::pos());
+    if (mousePosition.y() >= hover_label_height && mousePosition.x() <= width() / 2)
```
This doesn't seem to update the label position when mousing over an already-visible label:

https://github.com/user-attachments/assets/ab37c804-b309-48c8-8182-30562eaa827b

Would `if (m_hover_label->underMouse() && m_hover_label->x() == 0)` work better here?

Sidicer replied:
Thank you for the suggestion, works great

tcl3 commented:
Sorry it took me a while to get back to this. On my machine, the hover label currently overlaps the edge of the window - or the find in page widget, if visible.

![image](https://github.com/user-attachments/assets/7d0cf9d1-6f5f-4783-ae32-166f4513bce3)

If you could find a way to account for the size of the frame borders, that would be ideal. Alternatively you could move the hover label back to the position it was - slightly raised and inset from the bottom of the frame.

AtkinsSJ commented:
The maths for the label sizing seems very slightly off. I'm finding that the label is more than half the width of the window. When playing around on wikipedia.org with a very small window size, the label is able to flip to the opposite side, but remain under the cursor:

[Peek 2024-10-02 10-16 link label.webm](https://github.com/user-attachments/assets/87f34a45-d525-4891-8a4b-52e891df5de7)

It's not a deal-breaker, but it'd be good to make this 100% reliable. :^)

Sidicer commented:
> The maths for the label sizing seems very slightly off. I'm finding that the label is more than half the width of the window. When playing around on wikipedia.org with a very small window size, the label is able to flip to the opposite side, but remain under the cursor:

![ladybird-hoverlabel](https://github.com/user-attachments/assets/0521b0fa-0a72-45d1-aa50-def3ca52ef81)

I have fixed an issue that appeared between first and last commit which was to replace `QAbstractScrollArea` with `QWidget` and added a buffer of `10px` for the hover label size calculation

AtkinsSJ reviewed (approved):
Looks good to me. Let's get this merged finally! :tada:
